### PR TITLE
expand job init example with spread and affinity

### PR DESCRIPTION
**Pull request by preetapan**
Added spread and affinity to the longer job file in `nomad init`

### CODE REVIEW

File: command/job_init.go [hunk 1]
```diff
@@ -364,7 +410,7 @@ job "example" {
         }
       }
 
-      # The "artifact" stanza instructs Nomad to download an artifact from a
+# The "artifact" stanza instructs Nomad to download an artifact from a
```
Comment (cgbaker) [inline command/job_init.go]: intentional?

Comment (cgbaker) [inline command/job_init.go]: is this change intentional and/or desired?

File: command/job_init.go [hunk 2]
```diff
@@ -340,6 +340,52 @@ job "example" {
       size = 300
     }
 
+
```
Comment (cgbaker) [inline command/job_init.go]: the other stanzas use indention (for commented description and fields)

File: command/job_init.go [hunk 3]
```diff
@@ -340,6 +340,52 @@ job "example" {
...
+    #
+    #     https://www.nomadproject.io/docs/job-specification/affinity.html
+    #
+    # affinity {
+    # attribute specifies the name of a node attribute or metadata
+    #   attribute = "${node.datacenter}"
+    #
+    # value specifies the desired attribute value. In this example Nomad
+    # will prefer placement in the "us-west1" datacenter.
+    #   value  = "us-west1"
```
Comment (shantanugadgil) [inline command/job_init.go]: this seems to make the example GCP specific.

do you think "dc2"/"dc3" would be a simpler hint?

File: command/job_init.go [hunk 4]
```diff
@@ -340,6 +340,52 @@ job "example" {
...
+    #     https://www.nomadproject.io/docs/job-specification/spread.html
+    #
+    # spread {
+    # attribute specifies the name of a node attribute or metadata
+    # attribute = "${node.datacenter}"
+    #
+    # targets can be used to define desired percentages of allocations
+    # for each targeted attribute value.
+    #
+    #   target "us-east1" {
```
Comment (shantanugadgil) [inline command/job_init.go]: this seems to make the example GCP specific.
do you think "dc1"/"dc2" would be a simpler hint?

File: command/job_init.go (hunk 3 above)
Comment (preetapan) [reply command/job_init.go]: since its commented out I don't have a strong opinion, its meant to be an example.
